src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -17,15 +23,60 @@ const signEventOrPrompt = async (message: MessageSignEvent) => {
     };
   }
 
+  const enableMessage: MessageAllowanceEnable = {
+    origin: message.origin,
+    args: {
+      host: message.origin.host,
+    },
+    action: "public/webln/enable",
+  };
+
+  await enable(enableMessage);
+
+  const dbPermissions = await db.permissions
+    .where("host")
+    .equalsIgnoreCase(message.origin.host)
+    .toArray();
+
+  const hasPermission = dbPermissions
+    .filter((permission) => permission.enabled)
+    .map(({ method }) => method)
+    .includes("signMessage");
+
   try {
-    const response = await utils.openPrompt<{
-      confirm: boolean;
-    }>({
-      ...message,
-      action: "public/nostr/confirmSignMessage",
-    });
-    if (!response.data.confirm) {
-      throw new Error("User rejected");
+    if (!hasPermission) {
+      const response = await utils.openPrompt<{
+        confirm: boolean;
+        rememberPermission: boolean;
+      }>({
+        ...message,
+        action: "public/nostr/confirmSignMessage",
+      });
+
+      if (!response.data.confirm) {
+        throw new Error("User rejected");
+      }
+
+      if (response.data.rememberPermission) {
+        const matchingAllowance = await db.allowances
+          .where("host")
+          .equalsIgnoreCase(message.origin.host)
+          .first();
+        const allowanceId = matchingAllowance?.id;
+
+        if (!allowanceId) return; // type-guard only
+
+        const permission: DbPermission = {
+          createdAt: Date.now().toString(),
+          allowanceId,
+          host: message.origin.host,
+          method: "signMessage",
```

Comment:
I think this is wrong, this mus be scoped to nostr/ as we also scope the webln calls. 
cc @reneaaron

Reply:
I think we should create types for these and combine those with i.e. this: https://github.com/getAlby/lightning-browser-extension/pull/1752/files#diff-82b11e63afd4e5a1aeac4c0023524f3d17f0aafecac1a8017236bb8f210082d0R82
Mentioned this in the PR description

Reply:
We should type this, wdyt?: https://github.com/getAlby/lightning-browser-extension/blob/feat/1727_nostr-permission-sign-message/src/extension/background-script/actions/ln/request.ts/#L24

Reply:
no idea, for me those are then just looooong random list of strings. and the same values in a type as in an array.
if TypeScript wants it, do it.

Reply:
@lisabaut wdyt?

Reply:
As discussed in person:
Because right now this PR deals only with "signMessage" in one single file
and more Nostr events are later to come
I think it is fine to keep it like you started it: [PermissionMethodNostr TYPE](https://github.com/getAlby/lightning-browser-extension/pull/1775/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR487)

At some point you will need a "supportedNostrMethodMap" object and map the method to its corresponding translation. 
Similar to how it was started here for LN (=> [draft PR code](https://github.com/getAlby/lightning-browser-extension/pull/1752/files#diff-82b11e63afd4e5a1aeac4c0023524f3d17f0aafecac1a8017236bb8f210082d0R82-R108))